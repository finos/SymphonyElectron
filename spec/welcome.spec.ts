import { shallow } from 'enzyme';
import * as React from 'react';
import Welcome from '../src/renderer/components/welcome';
import { ipcRenderer } from './__mocks__/electron';

describe('welcome', () => {
  const welcomeLabel = 'welcome';
  const welcomeMock = {
    url: 'https://my.symphony.com',
    message: '',
    urlValid: true,
    isPodConfigured: false,
    isSeamlessLoginEnabled: true,
  };
  const onLabelEvent = 'on';
  const removeListenerLabelEvent = 'removeListener';

  it('should render correctly', () => {
    const wrapper = shallow(React.createElement(Welcome));
    expect(wrapper).toMatchSnapshot();
  });

  it('should call `welcome` event when component is mounted', () => {
    const spy = jest.spyOn(ipcRenderer, onLabelEvent);
    shallow(React.createElement(Welcome));
    expect(spy).toBeCalledWith(welcomeLabel, expect.any(Function));
  });

  it('should remove listener `welcome` when component is unmounted', () => {
    const spyMount = jest.spyOn(ipcRenderer, onLabelEvent);
    const spyUnmount = jest.spyOn(ipcRenderer, removeListenerLabelEvent);

    const wrapper = shallow(React.createElement(Welcome));
    expect(spyMount).toBeCalledWith(welcomeLabel, expect.any(Function));

    wrapper.unmount();
    expect(spyUnmount).toBeCalledWith(welcomeLabel, expect.any(Function));
  });

  it('should call `updateState` when component is mounted', () => {
    const spy = jest.spyOn(Welcome.prototype, 'setState');
    shallow(React.createElement(Welcome));

    ipcRenderer.send('welcome', welcomeMock);

    expect(spy).toBeCalledWith(welcomeMock);
  });

  it('should change pod url in text box', () => {
    const podUrlMock = {
      url: 'https://corporate.symphony.com',
      message: '',
      urlValid: true,
    };

    const spy = jest.spyOn(Welcome.prototype, 'setState');
    const updatePodUrlSpy = jest.spyOn(Welcome.prototype, 'updatePodUrl');

    const wrapper = shallow(React.createElement(Welcome));
    ipcRenderer.send('welcome', welcomeMock);

    const welcomePodUrlBox = `input.Welcome-main-container-podurl-box`;
    const input = wrapper.find(welcomePodUrlBox);

    input.simulate('focus');
    input.simulate('change', {
      target: { value: 'https://corporate.symphony.com' },
    });

    expect(updatePodUrlSpy).toBeCalled();
    expect(spy).toBeCalledWith(podUrlMock);
  });

  it('should show message for invalid pod url', () => {
    const podUrlMock = {
      url: 'abcdef',
      message: 'Please enter a valid url',
      urlValid: false,
    };

    const spy = jest.spyOn(Welcome.prototype, 'setState');
    const updatePodUrlSpy = jest.spyOn(Welcome.prototype, 'updatePodUrl');

    const wrapper = shallow(React.createElement(Welcome));
    ipcRenderer.send('welcome', welcomeMock);

    const welcomePodUrlBox = `input.Welcome-main-container-podurl-box`;
    const input = wrapper.find(welcomePodUrlBox);

    input.simulate('focus');
    input.simulate('change', { target: { value: 'abcdef' } });

    expect(updatePodUrlSpy).toBeCalled();
    expect(spy).toBeCalledWith(podUrlMock);
  });

  it('should set pod url', () => {
    const spy = jest.spyOn(Welcome.prototype, 'setState');
    const setPodUrlSpy = jest.spyOn(Welcome.prototype, 'login');

    const wrapper = shallow(React.createElement(Welcome));
    ipcRenderer.send('welcome', welcomeMock);
    const welcomeContinueButton = `button.Welcome-continue-button`;
    wrapper.find(welcomeContinueButton).simulate('click');

    expect(setPodUrlSpy).toBeCalled();
    expect(spy).toBeCalledWith(welcomeMock);
  });

  it('should not show pod url input field', () => {
    const welcomeMock = {
      url: 'https://my.symphony.com',
      message: '',
      urlValid: true,
      isPodConfigured: true,
      isSeamlessLoginEnabled: true,
    };
    const wrapper = shallow(React.createElement(Welcome));
    ipcRenderer.send('welcome', welcomeMock);
    const podUrlBox = `input.Welcome-main-container-podurl-box`;
    expect(wrapper.find(podUrlBox).getElements()).toEqual([]);
  });
});
