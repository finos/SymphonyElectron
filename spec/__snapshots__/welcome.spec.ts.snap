// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome should render correctly 1`] = `
<div
  className="Welcome"
  lang="en-US"
>
  <div
    className="Welcome-image-container"
  >
    <img
      alt="Symphony Logo"
      src="../renderer/assets/symphony-logo-plain.png"
    />
  </div>
  <div
    className="Welcome-main-container"
  >
    <h3
      className="Welcome-name"
    >
      Pod URL
    </h3>
    <div
      className="Welcome-main-container-input-div"
    >
      <div
        className="Welcome-main-container-input-selection"
      >
        <input
          className="Welcome-main-container-podurl-box"
          onChange={[Function]}
          type="url"
          value="https://[POD].symphony.com"
        />
      </div>
      <div
        className="Welcome-main-container-sso-box"
        title="Enable Single Sign On"
      >
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          SSO
        </label>
      </div>
    </div>
    <label
      className="Welcome-message-label"
    />
    <button
      className="Welcome-continue-button-disabled"
      disabled={true}
      onClick={[Function]}
    >
      Continue
    </button>
  </div>
</div>
`;
